less/scripts/train/warmup_lora_train.py

```diff
@@ -0,0 +1,91 @@
+import subprocess
+from pathlib import Path
+import argparse
+import random  
+
+def run_training(
+    train_file: str,
+    model_name_or_path: str, 
+    percentage: float=0.05,
+    data_seed: int=3,
+):
+    job_name = f"{model_name_or_path.split('/')[-1]}-p{percentage}-lora-seed{data_seed}"
+    output_dir = Path("../out") / job_name
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    # Base training args from base_training_args.sh
+    base_training_args = [
+    "--do_train", "True",
+    "--max_seq_length", "2048",
+    "--use_fast_tokenizer", "True",
+    "--lr_scheduler_type", "linear",
+    "--warmup_ratio", "0.03",
+    "--weight_decay", "0.0",
+    "--evaluation_strategy", "no",
+    "--logging_steps", "1",
+    "--save_strategy", "epoch",
+    "--num_train_epochs", "4",
+    "--bf16", "True",
+    "--tf32", "False",
+    "--fp16", "False",
+    "--overwrite_output_dir", "True",
+    "--report_to", "wandb",
+    "--optim", "adamw_torch",
+    "--lora", "True",
+    "--lora_r", "128",
+    "--lora_alpha", "512",
+    "--lora_dropout", "0.1",
+    "--lora_target_modules", "q_proj", "k_proj", "v_proj", "o_proj",
+    "--learning_rate", "2e-05",
+    "--per_device_train_batch_size", "1",
+    "--gradient_accumulation_steps", "32"
+]
+
+    # Add FSDP config for large models
+    if model_name_or_path == "meta-llama/Llama-2-13b-hf":
```

Comment:
This is the original code from their .sh script, right?

Reply:
Yes, it seems like models larger than llama-7B would not fit into GPUs so they needed to enable FSDP for larger experiments